Test/client_test.py

```diff
@@ -41,17 +40,22 @@ def test_index_page2(client):
     response = client.get('/')
     assert response.status_code == 302
 
-def test_index_page_with_mock(mock_get_requests, client):
+@patch('pymongo.MongoClient')
+def test_index_page_with_mock(mock_mongo_card, client):
+    mock_mongo_card.return_value["ToDo"]["All Items"].find = substitute_mongo_api_mock
     response = client.get('/items/get_all_cards')
     assert response.status_code == 200 
-    
-def test_index_page_with_mock2(mock_get_requests, client):
-    response = client.get('/items/get_all_cards')
-    data = Trello.get_all_cards_from_board(response)
-    data2 = json.loads(data)
-    assert "Test to do item" in data
-    assert "12345" in data
-    assert "67890" in data
-    assert "2020-08-01T12:52:06.278Z" in data
-    assert data2[0] == {'id': '12345', 'name': 'Test to do item', 'idList': '67890', 'dateLastActivity': '2020-08-01T12:52:06.278Z'}
-    assert data2[0]['name'] == "Test to do item"
\ No newline at end of file
+
+
+# Cant get below to work!! Get AssertionError assert <MagicMcok.... == {'_id':'1234...} HELP
+#@patch('pymongo.MongoClient')
+#def test_index_page_with_mock2(mock_mongo_card, client):
+#   mock_mongo_card.return_value["ToDo"]["All Items"].find = substitute_mongo_api_mock  # returns a cursor
+#   data = [TodoItem.from_mongo_card(card) for card in mock_mongo_card]    
+#   assert mock_mongo_card.return_value[0] == {'_id': '123456', 'title': 'Test to do item', 'update_time': '2020-08-01T12:52:06.278Z'}
+#   assert data[1]['title'] == "Test to do item"
+#   assert "Test to do item" in data
+#   assert "12345" in mock_mongo_card
+#   assert "2020-08-01T12:52:06.278Z" in mock_mongo_card
+#   assert data2[0] == {'_id': '12345', 'title': 'Test to do item', 'update_time': '2020-08-01T12:52:06.278Z'}
+#   assert mock_mongo_card[0]['title'] == "Test to do item"
```

Comment:
It's not totally clear here what exactly you're hoping to test here.

If you want to verify that the mocked information appears on the page, we could do that with something like
```suggestion
@patch('pymongo.MongoClient')
def test_index_page_with_mock2(mock_mongo_card, client):
    mock_mongo_card.return_value["ToDo"]["All Items"].find = substitute_mongo_api_mock
    response = client.get('/items/get_all_cards')
    assert "Test to do item" in response.data.decode()
```

If we wanted to test the mock itself directly to check it's working, then we'd need to call the mock directly, and access the data it as we do in the code - see below -  but this would then be us just testing the mock which is a bit pointless!
```
@patch('pymongo.MongoClient')
def test_index_page_with_mock2(mock_mongo_card, client):
   mock_mongo_card.return_value["ToDo"]["All Items"].find = substitute_mongo_api_mock  # returns a cursor
   #### E.g. access the mocked data with this line
   cards = mock_mongo_card()["ToDo"]["All Items"].find()
   data = [TodoItem.from_mongo_card(card) for card in cards]    
   assert data[0] == {'_id': '123456', 'title': 'Test to do item', 'update_time': '2020-08-01T12:52:06.278Z'}
```